torchgeo/datasets/utils.py

```diff
@@ -17,9 +17,11 @@
 from datetime import datetime, timedelta
 from typing import Any, TypeAlias, cast, overload
 
+import affine
```

Comment:
This is going to sound stupid, but can we import `rasterio.affine.Affine` instead? Rasterio vendors affine, so we don't officially need to add an additional dependency on affine.